static/js/millennium_diff_table.js

```diff
@@ -0,0 +1,541 @@
+ * @param {array} r4Resources - The array of R4 resource objects.
+ * @returns {object} Contains arrays of DSTU2 and R4 resources that match.
+ */
+function buildMatchFromConfig(config, dstu2Resources, r4Resources) {
+	let configuredResource = {
+		dstu2Resources: [],
+		r4Resources: [],
+	};
+
+	// Find the DSTU2 resources in the config
+	for (let i = 0; i < config.resourcesCombined.length; i++) {
+		for (let j = 0; j < dstu2Resources.length; j++) {
+			if (dstu2Resources[j] != null) {
+				if (dstu2Resources[j].type === config.resourcesCombined[i]) {
+					configuredResource.dstu2Resources.push(buildResource(dstu2Resources[j]));
+					delete dstu2Resources[j];
+				}
+			}
+		}
+	}
+
+	// Find the R4 resources in the config
+	for (let i = 0; i < config.combinedTo.length; i++) {
+		for (let j = 0; j < r4Resources.length; j++) {
+			if (r4Resources[j] != null) {
+				if (r4Resources[j].type === config.combinedTo[i]) {
+					configuredResource.r4Resources.push(buildResource(r4Resources[j]));
+					delete r4Resources[j];
+				}
+			}
+		}
+	}
+	
+	if (config.hasOwnProperty('notes')) {
+		configuredResource.notes = config.notes;
+	}
+	return configuredResource;
+}
+
+/**
+ * Matches the resources between DSTU2 and R4.
+ * @param {object} conformanceData - The DSTU2 metadata.
+ * @param {object} capabilityStatementData - The R4 metadata.
+ * @returns {array} Array of all matching resource objects.
+ */
+function matchResources(conformanceData, capabilityStatementData) {
+  let dstu2Resources = conformanceData.rest[0].resource;
+  let r4Resources = capabilityStatementData.rest[0].resource;
+	
+  let matchingResources = [];
+	let currentResource = null;
+	let config = null;
+	let match = null;
+
+	// Match all DSTU2 Resources
+  for (let i = 0; i < dstu2Resources.length; i++) {
+    currentResource = dstu2Resources[i];
+		
+		if (currentResource != null) {
+			config = getConfiguration(currentResource.type);
+			match = null;
+			
+			if (config != null) {
+				match = buildMatchFromConfig(config, dstu2Resources, r4Resources);
+				matchingResources.push(match);
+			}
+			else {
+				match = buildMatch(currentResource, r4Resources);
+				matchingResources.push(match);
+			}
+		}
+  }
+	
+	// Push the remaining R4 resources that were not in DSTU2
+	for (let i = 0; i < r4Resources.length; i++) {
+		currentResource = r4Resources[i];
+		
+		if (currentResource != null) {
+			config = getConfiguration(currentResource.type)
+			
+			if (config != null) {
+				// Build and push all resources extended from the Basic resource
+				for (let j = 0; j < config.length; j++) {
+					matchingResources.push({
+						r4Resources: [buildResource(config[j])],
+						notes: config[j].notes
+					});
+				}
+			}
+			else {
+				matchingResources.push({
+					r4Resources: [buildResource(currentResource)]
+				});
+			}
+		}
+	}
+	
+	return matchingResources;
+}
+
+/**
+ * Sorts the matching resources alphabetically, first by DSTU2 and then by R4.
```

Comment:
since R4 is the intended version, we should sort on it rather than have it match DSTU 2 and make another section of new APIs at the bottom

Reply:
Updated here: a93979d3122a40243f025ffd6ed8ceca5589c18a